src/components/Table/Table.jsx

```diff
@@ -68,6 +68,7 @@ const propTypes = {
   columns: TableColumnsPropTypes.isRequired,
   /** Row value data for the body of the table */
   data: TableRowPropTypes.isRequired,
+  oldData: PropTypes.arrayOf(PropTypes.object),
```

Comment:
I have a hard time understand what `oldData` is and how it is used.  Can you explain what `oldData` is doing here in `Table`?

Reply:
I have changed `oldData` to 'preData' and removed `oldData` in `Table`. `preData` is a state and means previous table data. It is used to save current table data after user clicked the edit icon. After the saving, table can do undo function when user clicked cancel button or the toast. It only shows up in `AsyncTable` and `tableReducer` now.